packages/typescript-estree/src/create-program/getScriptKind.ts

```diff
@@ -1,6 +1,32 @@
 import path from 'path';
 import * as ts from 'typescript';
 
+function isNewModuleExtension(filePath: string): boolean {
```

Comment:
I suggest a better name, because "new" doesn't age where. What about `isNodeNextModuleExtension`

Reply:
The reason why I didn't go with `Node16` is exactly because it's tied to a specific feature and if that resolution mode goes out of fashion (e.g. when `Node24` is introduced, etc.), the function would seem awkward. Unless you plan to make this function do exactly `Node16` logic and do not intend it to accomodate new extensions in the future, I suggest using something more fluid like `NodeNext`.

Reply:
That's fine - more just aware that "node-next" is something that changes meaning over time, but the method logic won't unless updated. Ultimately happy with either, so will make it `isNodeNextModuleExtension`.

Reply:
https://github.com/typescript-eslint/typescript-eslint/pull/6795/commits/5a5bc878a65e62682cba2c61cf8ccafc34b35e58